app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -79,28 +126,42 @@ private void storeStuct(IASTDeclaration[] declarations, String name) throws Exce
         structRepo.put(name, struct);
     }
 
-    private void associateIDStruct(IASTNode[] comments) {
-        Pattern pattern = Pattern.compile("@canPayloadStruct\\s+(\\S+)\\s*=\\s*(\\S+)");
+    private void parseComments(IASTNode[] comments) {
+        Pattern pattern1 = Pattern.compile("@canPayloadStruct\\s+(\\S+)\\s*=\\s*(\\S+)");
+        Pattern pattern2 = Pattern.compile("@payloadDataType\\s+(\\S+)\\s+=\\s*(\\S+)");
         for (IASTNode comment : comments) {
-            Matcher matcher = pattern.matcher(comment.getRawSignature());
-            if (matcher.find()) {
-                IDToStruct.put(matcher.group(1), matcher.group(2));
+            Matcher matcher1 = pattern1.matcher(comment.getRawSignature());
+            Matcher matcher2 = pattern2.matcher(comment.getRawSignature());
+            if (matcher1.find()) { // associate ID to struct
+                canStructRepo.put(matcher1.group(1), matcher1.group(2));
+            } else if (matcher2.find()) { // assoicate payload data type
+                getConst(matcher2.group(1)).payLoadDataType = matcher2.group(2);
```

Comment:
Can you merge your internal format for these two cases? Both are variants of "this CAN ID has a payload of some format", whether that format is a single data type (eg, uint32_t or float) or a struct. We can discuss how to architect this cleanly.